packages/flutter/lib/src/services/hardware_keyboard.dart

```diff
@@ -704,45 +731,89 @@ typedef KeyMessageHandler = bool Function(KeyMessage message);
+  /// All destinations will always receive the event regardless of the handlers'
+  /// results. If any handler's result is true, then the overall result of the
+  /// second phase is true.
+  ///
+  /// See also:
   ///
-  /// Key messages received from the platform are first sent to [RawKeyboard]'s
-  /// listeners and [HardwareKeyboard]'s handlers, then sent to
-  /// [keyMessageHandler], regardless of the results of [HardwareKeyboard]'s
-  /// handlers. The event results from the handlers and [keyMessageHandler] are
-  /// combined and returned to the platform. The event result is explained
-  /// below.
+  ///  * [RawKeyboard.addListener], which adds a raw keyboard listener.
+  ///  * [RawKeyboardListener], which is also implemented by adding a raw
+  ///    keyboard listener.
+  ///  * [HardwareKeyboard.addHandler], which adds a hardware keyboard handler.
   ///
-  /// For most common applications, which use [WidgetsBinding], this field
-  /// is set by the focus system (see `FocusManger`) on startup and should not
-  /// be change explicitly.
+  /// ## Advanced usages: Manual assignment or patching
   ///
   /// If you are not using the focus system to manage focus, set this
   /// attribute to a [KeyMessageHandler] that returns true if the propagation
   /// on the platform should not be continued. If this field is null, key events
-  /// will be assumed to not have been handled by Flutter.
-  ///
-  /// ## Event result
-  ///
-  /// Key messages on the platform are given to Flutter to be handled by the
-  /// engine. If they are not handled, then the platform will continue to
-  /// distribute the keys (i.e. propagate them) to other (possibly non-Flutter)
-  /// components in the application. The return value from this handler tells
-  /// the platform to either stop propagation (by returning true: "event
-  /// handled"), or pass the event on to other controls (false: "event not
-  /// handled"). Some platforms might trigger special alerts if the event
-  /// is not handled by other controls either (such as the "bonk" noise on
-  /// macOS).
-  ///
-  /// The result from [keyMessageHandler] and [HardwareKeyboard]'s handlers
-  /// are combined. If any of the handlers claim to handle the event,
-  /// the overall result will be "event handled".
+  /// will be assumed to not have been handled by Flutter, a result of "false".
+  ///
+  /// Even if you are using the focus system, you might also want to do more
+  /// than the focus system allows. In these cases, you can _patch_
+  /// [keyMessageHandler] by assigning it with a callback that performs your
+  /// tasks and calls the original callback in between (or not at all.)
+  ///
+  /// Patching [keyMessageHandler] can not be reverted. You should always assume
+  /// that other component might haved patch it before you and after you.
+  /// This means that you might want to write your own global notification
+  /// manager, which callbacks can be added to and removed from.
```

Comment:
```suggestion
  /// manager, to which callbacks can be added and removed.
```

Reply:
Is it correct, since you can't remove the callback "to" the manager?

Reply:
Yes, the "to" doesn't have to agree with both of the things in the list, just the first one. It's quirky: you wouldn't say "from which callbacks can be added and removed", but you could say "from which callbacks can be removed and added".